org.eclipse.ltk.core.refactoring/src/org/eclipse/ltk/internal/core/refactoring/UndoableOperation2ChangeAdapter.java

```diff
@@ -40,7 +43,7 @@
 import org.eclipse.ltk.core.refactoring.RefactoringStatus;
 
 
-public class UndoableOperation2ChangeAdapter implements IUndoableOperation, IAdvancedUndoableOperation  {
+public class UndoableOperation2ChangeAdapter implements IUndoableOperation, IAdvancedUndoableOperation, ISchedulableOperation  {
```

Comment:
I would like to know why we really needed the `ISchedulableOperation` interface here ?

Reply:
please see Bug 578871. Otherwise it will freeze the UI - as long as any other Thread (for example autobuild) has any lock in the workspace.

Reply:
Ok thanks @jukzi i didn't refer to the platform ui PR which is still open. The change make sense then. But shall we wait till we have the ui changes as well ? Otherwise the change LGTM

Reply:
can you please trigger rebuild?

Reply:
is there a way to update commit message within github? i dont find the button

Reply:
It is called "amend commit" and "force push" from Eclipse